vector_setters.go

```diff
@@ -32,108 +33,319 @@ func setPrimitive[T any](vec *vector, rowIdx C.idx_t, v T) {
 	xs[rowIdx] = v
 }
 
-func (vec *vector) setTS(t Type, rowIdx C.idx_t, val any) {
-	v := val.(time.Time)
+// TODO: support Decimal here and bool and hugeint
```

Comment:
Let's remove this comment and add a `FIXME` to the relevant `set...` functions. We can pick this up later.